text/XXXX-replace_publish_clone_with_push_pull.md

```diff
@@ -0,0 +1,160 @@
+- Feature Name: Dataset Push & Pull
+- Start Date: <!-- (fill me in with today's date, YYYY-MM-DD) -->
+- RFC PR: <!-- (leave this empty) -->
+- Issue: <!-- (leave this empty) -->
+
+# Summary
+[summary]: #summary
+
+Replace `publish/add` with `push/pull`, which default to pushing all data, pulling only the latest version. A new `clone` command combines `pull` with `checkout`.
```

Comment:
Discussed in a recent meeting, but just repeating here: I think we can avoid adding a `clone` command by just letting `checkout` just pull the data first if the user is connected.